docs/developer/architecture/configurations.rst

```diff
@@ -91,17 +91,36 @@ requirements for the following reasons:
 How it works
 ^^^^^^^^^^^^
 
-During the Bootstrap stages, when we are assertive that the K8s cluster is
-fully ready and available we could perform the following actions:
-
-  - Create and deploy ConfigMaps that hold cluster and service configurations
-    and pre-fill them with default values.
-  - Template service pods and deployments to consume configuration data
-    directly from the above deployed ConfigMaps
+During Bootstrap, Upgrade or Downgrade stages, when we are assertive that
+the K8s cluster is fully ready and available we could perform the following
+actions:
+
+  - Firstly, create and deploy ConfigMaps that will hold customizable cluster
+    and service configurations.
+    These ConfigMaps should define an empty `config.yaml` in the data section
+    of the ConfigMap for later use. A standard layout for each field
+    could be added in the documentation to assist MetalK8s administrator in
+    adding and modifying customizations.
+  - In the `defaults.yaml` located in `salt/metalk8s` directory,
+    define the keys and values for default service configurations in a
+    structured format.
+
+      - The layout of service configurations within this file could have root
+        key being the impacted ConfigMap name.
+  - During Addon manifests rendering, call a Salt module that will merge
+    the configurations defined within the customizable ConfigMap to those
+    defined in `defaults.yaml` using the recommended merge strategy.
```

Comment:
As I discussed with you @Ebaneck, it's not a good idea to have this default value in `defaults.yaml` if you do not use the pillar to merge it with.

For me you should have a `csc.yaml` (or another name) just next to each CSC salt state (e.g.: for dex `salt/metalk8s/addons/dex/deployed/csc.yaml`) containing all the defaults value for the config and in the SLS you just load this yaml file and get + merge with the associated config map.

Reply:
@TeddyAndrieux your comments have been taken into consideration but I prefer to have a single `csc.yaml` found at the root of `salt/metalk8s`

Reply:
Then what's the difference between a key in `defaults.yaml` and a single `csc.yaml` file? I agree with Teddy's point, maintaining these defaults close to where we will use them makes it simpler to develop (and also, if you need to add/remove addons, you won't need to care about which parts of some common files you need to change, just create/delete a new directory in `salt/metalk8s/addons` and be done with it)

Reply:
ok, I see